pkg/frontend/frontend.go

```diff
@@ -201,29 +226,8 @@ func (f *frontend) Run(ctx context.Context, stop <-chan struct{}, done chan<- st
 		}()
 	}
 
-	r := mux.NewRouter()
-	r.Use(middleware.Log(f.baseLog))
-	r.Use(middleware.Metrics(f.m))
-	r.Use(middleware.Panic)
-	r.Use(middleware.Headers(f.env))
-	r.Use(middleware.Validate(f.env, f.apis))
-	r.Use(middleware.Body)
-
-	r.NotFoundHandler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		w.Header().Set("Content-Type", "application/json")
-		api.WriteError(w, http.StatusNotFound, api.CloudErrorCodeNotFound, "", "The requested path could not be found.")
-	})
-	r.NotFoundHandler = middleware.Authenticated(f.env)(r.NotFoundHandler)
-
-	unauthenticated := r.NewRoute().Subrouter()
-	f.unauthenticatedRoutes(unauthenticated)
-
-	authenticated := r.NewRoute().Subrouter()
-	authenticated.Use(middleware.Authenticated(f.env))
-	f.authenticatedRoutes(authenticated)
-
 	f.s = &http.Server{
-		Handler:      middleware.Lowercase(r),
+		Handler:      middleware.Lowercase(f.setupRouter()),
```

Comment:
nit for next time, might be worth getting middleware.Lowercase() in setupRouter too.

Reply:
@jim-minter yeah, I was thinking about it, but decided to keep it here, because I wasn't sure why it wasn't in router itself initially. My guess was: a middleware added like `r.Use(middleware.Lowercase)` will only be called on a route match, but we might want lowercase 404s.